### PR TITLE
feat: Add convert to model tests for all node types

### PR DESCRIPTION
**Pull request by biblicalph**
Add ConvertToTerraformModel tests for all nodes.
Fail tests if a node type does not have tests for convert to model.
A node should provide the test by creating a JSON file in the testdata directory under the specific node's type - source, procesor or destination.
The file name should be the underscored value of the type name with which the node was registered.
Example: [mezmo logs sink](https://github.com/mezmo/terraform-provider-mezmo/blob/cd3af40e56fbda71c253bc25b9699eb3e6d77fd0/internal/provider/destination_resource_definitions.go#L111) has type `logs`. A file named `logs.json` is thus created for it in `destinations` directory

Ref: [LOG-18469](https://mezmo.atlassian.net/browse/LOG-18469)

### CODE REVIEW

File: internal/provider/models/destinations/http.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
 	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
```
Comment (darinspivey) [inline internal/provider/models/destinations/http.go:12]: This looks like a dupe, but not exported to `.`. It should be one or the other.

File: internal/provider/models/destinations/loki.go [hunk 2]
```diff
@@ -159,7 +165,11 @@ func LokiDestinationToModel(plan *LokiDestinationModel, component *Destination)
 				value := obj["label_value"].(string)
 				labelMap[key] = value
 			}
-			plan.Labels = basetypes.NewMapValueMust(plan.Labels.ElementType(nil), MapAnyToMapValues(labelMap))
+			labelType := plan.Labels.ElementType(context.TODO())
```
Comment (darinspivey) [inline internal/provider/models/destinations/loki.go:168]: TIL, what's `context.TODO`, and why this versus the `Background()` one ?

File: internal/provider/models/destinations/test/elasticsearch_test.go [hunk 3]
```diff
@@ -72,7 +72,7 @@ func TestElasticSearchDestinationResource(t *testing.T) {
 						description = "my destination description"
 						inputs      = [mezmo_http_source.my_source.id]
 						pipeline_id = mezmo_pipeline.test_parent.id
-						endpoints   = ["https://example.com"]
+						endpoints   = ["https://google.com"]
```
Comment (darinspivey) [inline internal/provider/models/destinations/test/elasticsearch_test.go:75]: Curious why this change.  `example.com` is a real domain after all.

File: pkg/resources/convertible_test.go [hunk 4]
```diff
@@ -656,3 +607,123 @@ func TestRouteResource(t *testing.T) {
...
+		convResTy := reflect.TypeOf(convRes).String()
+		if _, ok := unmatchedResources[convResTy]; ok {
+			delete(unmatchedResources, convResTy)
+		} else {
+			unmatchedResources[convResTy] = true
+		}
+	}
+
+	for unmatchedType, which := range unmatchedResources {
+		if which == false {
```
Comment (darinspivey) [inline pkg/resources/convertible_test.go:701]: nit: can `which` have a better name?  It feels more like `success` in this context, but the name `unmatchedType` already implies a failure.

File: pkg/resources/testdata/destinations/azure_blob_storage.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "compression": "gzip",
+    "container_name": "logs",
+    "connection_string": "http://localhost:3000",
+    "prefix": "error_"
+  },
+  "inputs": [
+    "ad6ebb62-63ae-11ee-9b45-26dab184329f",
+    "79551faa-6c6a-11ee-be81-6671faf7df66"
+  ]
+}
```
Comment (darinspivey) [inline pkg/resources/testdata/destinations/azure_blob_storage.json:23]: Most of these are missing a trailing newline.  You can make your editor always add one upon save...

**Review by darinspivey — COMMENTED** [review]
Looks fine with a few suggestions.  I'm new to the export stuff, so I'll defer approval to @dhable

File: internal/provider/models/destinations/loki.go (hunk 2 above)
Comment (biblicalph) [reply internal/provider/models/destinations/loki.go:168]: ElementType does not use the context it is passed.
I do have go static checker which advises to pass a TODO context in such cases: https://staticcheck.io/docs/checks#SA1012

File: internal/provider/models/destinations/test/elasticsearch_test.go (hunk 3 above)
Comment (biblicalph) [reply internal/provider/models/destinations/test/elasticsearch_test.go:75]: The domain is down hardddd

File: pkg/resources/testdata/destinations/azure_blob_storage.json (hunk 5 above)
Comment (biblicalph) [reply pkg/resources/testdata/destinations/azure_blob_storage.json:23]: It's a JSON file - trailing commas are not allowed

File: internal/provider/models/destinations/loki.go (hunk 2 above)
Comment (darinspivey) [reply internal/provider/models/destinations/loki.go:168]: ah nice, thanks!

File: pkg/resources/testdata/destinations/azure_blob_storage.json (hunk 5 above)
Comment (darinspivey) [reply pkg/resources/testdata/destinations/azure_blob_storage.json:23]: really?!  I don't think I've ever encountered that, but ok.

Comment (jmoses) [reply pkg/resources/testdata/destinations/azure_blob_storage.json:23]:
```
➜  ~  (⎈ |a2.us-east-1.stage.logdna:default)
> echo '["a",]' | jq
parse error: Expected another array element at line 1, column 6
➜  ~  (⎈ |a2.us-east-1.stage.logdna:default)
> echo '["a"]' | jq
[
  "a"
]
```

File: internal/provider/models/destinations/http.go (hunk 1 above)
Comment (dhable) [reply internal/provider/models/destinations/http.go:12]: Yes, I'd stick to one import style or the other. Using both is a bit confusing.

File: pkg/resources/testdata/destinations/azure_blob_storage.json (hunk 5 above)
Comment (darinspivey) [reply pkg/resources/testdata/destinations/azure_blob_storage.json:23]: yeah, I think that a newline that's outside of the JSON doesn't matter

Comment (darinspivey) [reply pkg/resources/testdata/destinations/azure_blob_storage.json:23]: OH!  not "trailing commas", trailing newlines--to keep github from printing the red "NO NEWLINE" warning

File: internal/provider/models/destinations/loki.go (hunk 2 above)
Comment (dhable) [reply internal/provider/models/destinations/loki.go:168]: I found [this response](https://www.mongodb.com/community/forums/t/go-why-context-todo-and-not-context-background/5009/3) regarding the use in MongoDB drivers helpful:

> As you mentioned in your response, they are the same context (no timeout/deadline and no associated values). The difference is semantic. I think of context.Background as “I’m deliberately passing in an empty context” and context.TODO as “there should be some other context here, which could be empty, but I’m not sure what the right value is yet so here’s a placeholder”.
>
> We use context.TODO() in the examples embedded in our documentation because the correct value would depend on the calling function, which is usually code from the user’s application. In this case, it’s up to the user to decide what the context should be, which usually depends on things like the maximum time they want the operation to take and whether or not the calling function has a context that it can propagate.

Since we're sure that we need a nil context at this code based on authoring the provider, we should probably sick with `context.Background()` every time we need a nil context object. Using `TODO()` makes it seem like we're not sure or that the implementation is incomplete.

File: internal/provider/models/destinations/test/elasticsearch_test.go (hunk 3 above)
Comment (dhable) [reply internal/provider/models/destinations/test/elasticsearch_test.go:75]: Stupid RFC fact of the day:

[RFC 2606](https://www.rfc-editor.org/rfc/rfc2606.html) defines four TLD that are reserved and avoid network traffic if used - `.test`, `.example`, `.invalid` and `.localhost`. Maybe we should consider using `.test` if we're worried about the time it takes to resolve to an IP address or some other network activity.

**Review by dhable — APPROVED** [review]
+1
